ui/app/src/views/ViewDashboardList.tsx

```diff
@@ -16,19 +16,64 @@ import {
   AccordionDetails,
   AccordionSummary,
   Box,
+  Button,
   CircularProgress,
   Container,
+  IconButton,
   Stack,
   Typography,
 } from '@mui/material';
+
+import { MouseEvent } from 'react';
 import { DashboardResource } from '@perses-dev/core';
 import { ErrorAlert, ErrorBoundary } from '@perses-dev/components';
 import ChevronDown from 'mdi-material-ui/ChevronDown';
 import FolderPound from 'mdi-material-ui/FolderPound';
+import TrashCan from 'mdi-material-ui/TrashCan';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import { useDashboardList } from '../model/dashboard-client';
+import { useCreateProjectMutation, useDeleteProjectMutation, ProjectModel } from '../model/project-client';
+import { useSnackbar } from '../context/SnackbarProvider';
 import DashboardList from '../components/DashboardList';
+import ConfirmDialog from '../components/ConfirmDialog/ConfirmDialog';
+import CreateProjectDialog from '../components/CreateProjectDialog/CreateProjectDialog';
+
+class ConfirmDialogState {
+  open = false;
+  name: string | undefined = undefined;
+}
 
 function RenderDashboardList() {
+  const [deleteProjectConfirmState, setDeleteProjectConfirmState] = useState<ConfirmDialogState>({
```

Comment:
I think you can also infer if the delete project dialog is open by checking if name is defined or not. When you close the dialog, you can just set the name to undefined

Reply:
Interesting idea, I transformed it into useState<string> but then I have back the issue I wanted to solve which is the title containing briefly "undefined" before being closed. 

![image](https://user-images.githubusercontent.com/10258608/214269214-8730626a-ec10-4d06-8a23-2ff5c3469f45.png)
![image](https://user-images.githubusercontent.com/10258608/214269282-cd0fb521-5a48-4905-a6c6-be104d50de25.png)
![image](https://user-images.githubusercontent.com/10258608/214269327-134514f1-3426-4255-95f3-3a6d5a5d3767.png)

Reply:
Let me push the code in a separate commit for you to review

Reply:
Another proposal https://github.com/perses/perses/pull/916/commits/5602cd30edb4d9a90e2fbf617891ca7afbc18612
We still have two fields but at least there's not anymore this cumbersome code to reset but not reset the name.

This
```ts
  const closeDeleteProjectConfirmDialog = () => {
    // Force reuse of previous name to prevent from seeing "undefined" right before the dialog close
    setDeleteProjectConfirmState({ open: false, name: deleteProjectConfirmState.name });
  };
```
becomes this 
```ts
  const closeDeleteProjectConfirmDialog = () => {
    setOpenDeleteProjectDialog(false);
  };
```

Reply:
I set it as resolved as to me the last proposal is a good enough compromise. Feel free to reopen if you still want to discuss about that